File: app/src/main/java/com/dialogapp/dialog/model/MicroBlogResponse.java
package com.dialogapp.dialog.model;

import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.squareup.moshi.Json;

import java.util.List;

@Entity(tableName = "microblogData", primaryKeys = "feedUrl")
public class MicroBlogResponse {

    @Json(name = "title")
    public final String title;
    @Json(name = "home_page_url")
    public final String homePageUrl;
    @NonNull
    @Json(name = "feed_url")
    public final String feedUrl;
    @Json(name = "_microblog")
    @Embedded(prefix = "microblog_")
    public final Microblog microblog;
    @Nullable
    @Json(name = "author")
    @Embedded(prefix = "author_")
    public final Author author;
    @Json(name = "items")
    @Ignore
    public List<Item> items;

    public MicroBlogResponse(String title, String homePageUrl, String feedUrl, Microblog microblog,
                             @Nullable Author author) {
        this.title = title;
        this.homePageUrl = homePageUrl;
        this.feedUrl = feedUrl;
        this.microblog = microblog;
        this.author = author;
    }

    public static class Microblog {

        @Nullable
        @Json(name = "id")
        public final String id;
        @Nullable
        @Json(name = "username")
        public final String username;
        @Nullable
        @Json(name = "bio")
        public final String bio;
        @Nullable
        @Json(name = "is_following")
        public final Boolean is_following;
        @Nullable
        @Json(name = "is_you")
        public final Boolean is_you;
        @Nullable
        @Json(name = "following_count")
        public final Integer following_count;

        public Microblog(@Nullable String id, @Nullable String username, @Nullable String bio,
                         @Nullable Boolean is_following, @Nullable Boolean is_you, @Nullable Integer following_count) {
            this.id = id;
            this.username = username;
            this.bio = bio;
            this.is_following = is_following;
            this.is_you = is_you;
            this.following_count = following_count;
        }
    }
}

File: app/src/main/java/com/dialogapp/dialog/model/MicroBlogResponse.kt
package com.dialogapp.dialog.model

import androidx.annotation.NonNull
import androidx.annotation.Nullable
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.Ignore
import com.squareup.moshi.Json

@Entity(tableName = "microblogData", primaryKeys = ["feedUrl"])
data class MicroBlogResponse(@field:Json(name = "title")
                             val title: String,
                             @field:Json(name = "home_page_url")
                             val homePageUrl: String,
                             @field:NonNull
                             @field:Json(name = "feed_url")
                             val feedUrl: String,
                             @field:Json(name = "_microblog")
                             @field:Embedded(prefix = "microblog_")
                             val microblog: Microblog,
                             @param:Nullable
                             @field:Nullable
                             @field:Json(name = "author")
                             @field:Embedded(prefix = "author_")
                             val author: Author,
                             @field:Json(name = "items")
                             val posts: List<Post>) {

    data class Microblog(@param:Nullable
                         @field:Nullable
                         @field:Json(name = "id")
                         val id: String,
                         @param:Nullable
                         @field:Nullable
                         @field:Json(name = "username")
                         val username: String,
                         @param:Nullable
                         @field:Nullable
                         @field:Json(name = "bio")
                         val bio: String,
                         @param:Nullable
                         @field:Nullable
                         @field:Json(name = "is_following")
                         val is_following: Boolean?,
                         @param:Nullable
                         @field:Nullable
                         @field:Json(name = "is_you")
                         val is_you: Boolean?,
                         @param:Nullable
                         @field:Nullable
                         @field:Json(name = "following_count")
                         val following_count: Int?)
}
